tools/check-region-routing.py

```
@@ -0,0 +1,322 @@
+            return url
+    raise CheckError(f'no <string name="{REGIONS_URL_RESOURCE_NAME}"> in {resource_path}')
+
+
+def fetch(url, data=None, content_type=None):
+    """GET (or POST, with `data`) `url`, returning (status, body-text). Raises on transport failure."""
+    headers = {"User-Agent": USER_AGENT}
+    if content_type:
+        headers["Content-Type"] = content_type
+    request = urllib.request.Request(url, data=data, headers=headers)
+    try:
+        with urllib.request.urlopen(request, timeout=TIMEOUT_SECONDS, context=SSL_CONTEXT) as response:
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)